nemo/collections/nlp/data/datasets/sgd_dataset/data_processor.py

```diff
@@ -132,9 +146,27 @@ def get_dialog_examples(self, dataset):
             )
         dial_file = self.dial_files[(self._task_name, dataset)]
         logging.info(f"Loading dialogue examples from {dial_file}.")
+
+        if not os.path.exists(self.slots_relation_file):
+            logging.error(f"The processed dialogue file ({dial_file}) does not exist.")
         with open(dial_file, "rb") as f:
             dial_examples = np.load(f, allow_pickle=True)
             f.close()
+
+        if not os.path.exists(self.slots_relation_file):
+            logging.error(
+                f"Slots relation file {self.slots_relation_file} does not exist. It disables the carry-over mechanism of state tracker for switches between services."
+            )
+            slots_relation_list = {}
+        else:
+            with open(self.slots_relation_file, "rb") as f:
+                slots_relation_list = pickle.load(f)
+            logging.info(
+                f"Loaded the slot relation list for value carry-over between services from {self.slots_relation_file}."
+            )
+
+        self.schema_emb_processor.schemas.slots_relation_list = slots_relation_list
+
```

Comment:
maybe add an `update_slots_relations_list` method to the schema_emb_processor? why is it part of the schema_emb_proc not the dial_processor?

Reply:
It can be there as it needs to process the dialogues and it need the information created deep down the creation of dialogues. If I move it there, then all the functions for creating dialogues should get copied there too. Actually tried it once but not feasible. It doubles our dialogue processing time.

Reply:
ok, could you just add a setter method to the schema_processor then?

Reply:
replace with a method please

Reply:
Done.